File: bls_jztc/jztc/src/api/stats.ts
import request from '../utils/request';

/**
 * 统计数据返回接口
 */
export interface StatsData {
  clientCount: number;   // 客户数量
  exchangeCount: number; // 兑换次数
  publishCount: number;  // 发布数量
  revenueAmount: number; // 收益金额
}

/**
 * 趋势数据类型
 */
export interface TrendData {
  labels: string[];  // 日期标签
  values: number[];  // 当前选择的数据类型的值
  allValues: {       // 所有数据类型的值
    clients: number[];    // 客户数量
    exchanges: number[];  // 兑换数量
    publishes: number[];  // 发布数量
    revenue: (number|string)[];  // 收益金额
  };
  dataType: 'clients' | 'exchanges' | 'publishes' | 'revenue' | 'all';  // 数据类型
  period: 'week' | 'month' | 'year';  // 周期
}

/**
 * 获取仪表盘统计数据
 * @param periodType 周期类型：week-本周，month-本月，year-本年
 * @returns 统计数据
 */
export async function getStatsData(periodType: 'week' | 'month' | 'year') {
  return request.get<any, { data: StatsData }>('/stats/data', {
    params: { periodType }
  });
}

/**
 * 获取趋势分析数据
 * @param periodType 周期类型：week-本周，month-本月，year-本年
 * @param dataType 数据类型：clients-客户数量，exchanges-兑换数量，publishes-发布数量，revenue-收益金额，all-所有数据
 * @returns 趋势数据
 */
export async function getTrendData(
  periodType: 'week' | 'month' | 'year',
  dataType: 'clients' | 'exchanges' | 'publishes' | 'revenue' | 'all' = 'all'
) {
  return request.get<any, { data: TrendData }>('/stats/trend', {
    params: { periodType, dataType }
  });
} 

File: bls_jztc/demo/api/backend/v1/stats.go
package v1

import (
	"github.com/gogf/gf/v2/frame/g"
)

// 统计数据请求
type StatsDataReq struct {
	g.Meta     `path:"/stats/data" method:"get" tags:"统计数据" summary:"获取统计数据" security:"Bearer" description:"获取关键统计数据，包括注册客户数量、兑换数量、发布数量、收益金额"`
	PeriodType string `d:"week" json:"periodType" dc:"周期类型：week-本周，month-本月，year-本年"`
}

// 统计数据响应
type StatsDataRes struct {
	g.Meta        `mime:"application/json" example:"json"`
	ClientCount   int     `json:"clientCount" dc:"注册客户数量"`
	ExchangeCount int     `json:"exchangeCount" dc:"兑换数量"`
	PublishCount  int     `json:"publishCount" dc:"发布数量"`
	RevenueAmount float64 `json:"revenueAmount" dc:"收益金额"`
}

// 趋势分析请求
type StatsTrendReq struct {
	g.Meta     `path:"/stats/trend" method:"get" tags:"统计数据" summary:"获取趋势分析数据" security:"Bearer" description:"获取趋势分析数据，支持按周期类型查询"`
	PeriodType string `d:"week" json:"periodType" dc:"周期类型：week-本周，month-本月，year-本年"`
	DataType   string `v:"required#数据类型不能为空" json:"dataType" dc:"数据类型：clients-客户数量，exchanges-兑换数量，publishes-发布数量，revenue-收益金额，all-所有数据"`
}

// 趋势分析响应
type StatsTrendRes struct {
	g.Meta    `mime:"application/json" example:"json"`
	Labels    []string                 `json:"labels" dc:"时间标签"`
	Values    []interface{}            `json:"values" dc:"数据值"`
	AllValues map[string][]interface{} `json:"allValues" dc:"所有类型的数据值，仅在dataType为all时使用"`
	DataType  string                   `json:"dataType" dc:"数据类型"`
	Period    string                   `json:"period" dc:"周期类型"`
}
